homework45/src/ToDoList.jsx

```diff
@@ -0,0 +1,18 @@
+import { ToDoItem} from "./ToDoItem"
+import { useSelector, useDispatch } from 'react-redux'
+import { actions } from './store/index';
+
+export function ToDoList() {
+    const todos = useSelector(state => state.todos)
+
+    return (
+        <ul className="list">
+            {todos.length === 0 && "No Todos"}
+            {todos.map(((todo) => {
```

Comment:
```
{todos.map(((todo) => <ToDoItem id={todo.id} key={todo.id}/>))}
```

Reply:
if you pass full todo as a prop, then you don't need to call additionally `useSelector` inside

```
<ToDoItem todo={todo} key={todo.id}/>

// ...

function ToDoItem(props) {
 ~~const todo = useSelector(state => state.todos.find(obj => obj.id === id));~~
  const { todo } = props
}
```